bokeh/plotting/helpers.py

```diff
@@ -170,7 +171,33 @@ def _update_legend(plot, glyph_renderer):
         legend = legends[0]
     else:
         raise RuntimeError("Plot %s configured with more than one legend renderer" % plot)
-    legend.legends.append(glyph_renderer)
+
+    label = glyph_renderer.glyph.label
+
+    if plot._compound_legend is True:
```

Comment:
`is True` is unnecessary. `bool` has only two values, so `if plot._compund_glyph:` is sufficient.